### PR TITLE
Connected Angular App to MongoDB

### LINKED ISSUE #13: Connect the Angular App with MongDB through nodeJS
## Description
As a contributor,
I need set up the basic MEAN stack repository,
so that I can create the slack inviter on top of it.

### Update [Required]
- [ ] server.js
- [ ] common.service.spec.ts
- [ ] common.service.ts

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours 

@ancyp @Janiceilene Can you assign me the issue?

### PR DESCRIPTION
**Pull request by BaaniLeen**
### Description
As a contributor,
I need set up the basic MEAN stack repository,
so that I can create the slack inviter on top of it.

Fixes #13 

Can be merged after #15 

## Time Taken
3 hours 

Commands to be run to test it locally.

`npm install -g @angular/cli`
`npm install`
`npm install express --save`
`npm install mongoose -- save`
`npm install body-parser --save`
`npm install --save rxjs@6 rxjs-compat@6`
Now open two terminals. In one run the command `node server.js` and in the other `ng serve -o`

### Type of Change:
- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
I ran it locally. 

### Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

@ancyp @bethanyr @Janiceilene Can you please review the PR? Thanks! :)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+var express = require('express');
```
[the remark this answers is not in the document]
Comment (bethanyr) [reply server.js]: I posted in slack asking about the Strings must use doublequote feedback - are we following that for this project?

**Comment by ancyp** [comment]
Why do I get 
zone.js:2969 GET http://localhost:8080/api/getUser/ 404 (Not Found) on my localhost:4200

If we are putting the DB URL openly visible in the code, how are we doing authentication/ensuring security?

**Comment by BaaniLeen** [comment]
@ancyp I am trying to understand how express routing and the GET POST requests work, why that URL is not working, how to access that URL within the code. If you have any prior knowledge about it, it would be great if you could guide me about the same.

Regarding the security angle, I talked about it to @ExactlyMay , she said we will be integrating the final DB ID and credentials in the Travis CI, to ensure the security. And this will be taken up further in the All Teams Meeting. :)

**Comment by BaaniLeen** [comment]
@ancyp Also, if you want to try out the GET and POST requests you can navigate to the 'form' route and enter and delete data and see the reflection in the mLab DB. :)

**Comment by bethanyr** [comment]
@BaaniLeen - is this PR still needed?

**Comment by BaaniLeen** [comment]
@bethanyr No. Since, as discussed on slack we are not having an Angular App for the slack Inviter, so we can close this PR.
Link to the discussion :  https://systers-opensource.slack.com/archives/C99N86T43/p1532013151000468

**Pull request closed without merging**